instrumentation/opentelemetry-instrumentation-openai-v2/src/opentelemetry/instrumentation/openai_v2/utils.py

```diff
@@ -43,69 +48,138 @@ def wrapper(*args, **kwargs):
+            if arguments:
+                tool_call_dict["function"]["arguments"] = arguments.replace(
+                    "\n", ""
+                )
+
         calls.append(tool_call_dict)
     return calls
 
 
-def set_event_prompt(span, prompt):
-    span.add_event(
-        name="gen_ai.content.prompt",
-        attributes={
-            GenAIAttributes.GEN_AI_PROMPT: prompt,
-        },
+def set_server_address_and_port(client_instance, attributes):
+    base_client = getattr(client_instance, "_client", None)
+    base_url = getattr(base_client, "base_url", None)
+    if not base_url:
+        return
+
+    port = -1
+    if isinstance(base_url, URL):
+        attributes[ServerAttributes.SERVER_ADDRESS] = base_url.host
+        port = base_url.port
+    elif isinstance(base_url, str):
+        try:
+            url = urlparse(base_url)
+            attributes[ServerAttributes.SERVER_ADDRESS] = url.hostname
+            port = url.port
+        except Exception:
+            pass
+
+    if port and port != 443 and port > 0:
+        attributes[ServerAttributes.SERVER_PORT] = port
+
+
+def get_property_value(obj, property_name):
+    if isinstance(obj, dict):
+        return obj.get(property_name, None)
+
+    return getattr(obj, property_name, None)
+
+
+def message_to_event(message, span_ctx):
+    attributes = {
+        GenAIAttributes.GEN_AI_SYSTEM: GenAIAttributes.GenAiSystemValues.OPENAI.value
+    }
+    role = get_property_value(message, "role")
+    content = get_property_value(message, "content")
+
+    body = {}
+    if content:
+        body["content"] = content
+    if role == "assistant":
+        tool_calls = extract_tool_calls(message)
+        if tool_calls:
+            body = {"tool_calls": tool_calls}
+    elif role == "tool":
+        tool_call_id = get_property_value(message, "tool_call_id")
+        if tool_call_id:
+            body["id"] = tool_call_id
+
+    return Event(
+        name=f"gen_ai.{role}.message",
+        attributes=attributes,
+        body=body,
+        trace_id=span_ctx.trace_id,
+        span_id=span_ctx.span_id,
+        trace_flags=span_ctx.trace_flags,
+    )
+
+
+def choice_to_event(choice, span_ctx):
+    attributes = {
+        GenAIAttributes.GEN_AI_SYSTEM: GenAIAttributes.GenAiSystemValues.OPENAI.value
+    }
+
+    body = {
+        "index": choice.index,
+        "finish_reason": choice.finish_reason or "error",
+    }
+
+    if choice.message:
+        message = {
+            "role": choice.message.role
+            if choice.message and choice.message.role
+            else "assistant"
+        }
+        tool_calls = extract_tool_calls(choice.message)
+        if tool_calls:
+            message["tool_calls"] = tool_calls
+        content = get_property_value(choice.message, "content")
+        if content:
+            message["content"] = content
+        body["message"] = message
+
+    return Event(
+        name="gen_ai.choice",
```

Comment:
Nit: Should we have a collective place where we can refer to different event names (constants.py?) or might be good to add to sem conv.